File: 45. Jump Game II/45. Jump Game II.go
func jump(nums []int) int {
    if len(nums) <= 1 {
        return 0
    }
    
    jumps := 0
    currentMax := 0
    nextMax := 0
    
    for i := 0; i < len(nums) - 1; i++ {
        if i + nums[i] > nextMax {
            nextMax = i + nums[i]
        }
        
        if i == currentMax {
            jumps++
            currentMax = nextMax
            
            if currentMax >= len(nums) - 1 {
                break
            }
        }
    }
    
    return jumps
}

File: 45. Jump Game II/45. Jump Game II.cs
public class Solution {
    public int Jump(int[] nums) {
        if (nums.Length <= 1) return 0;
        
        int jumps = 0;
        int currentMax = 0;
        int nextMax = 0;
        
        for (int i = 0; i < nums.Length - 1; i++) {
            nextMax = Math.Max(nextMax, i + nums[i]);
            
            if (i == currentMax) {
                jumps++;
                currentMax = nextMax;
                
                if (currentMax >= nums.Length - 1) {
                    break;
                }
            }
        }
        
        return jumps;
    }
}